src/board-explorer-app/board-explorer-app.html

```diff
@@ -962,23 +942,47 @@
           /* Enable/Disable 'Categories' based on whether there are categories */
           this.$.filterControls.querySelector('[name="categories"]').disabled = (this.categories.length == 0);
 
+          this.fetching = true;
           /* If there are categories for this board, select the 'Categories' radio box */
-          if (this.categories.length) {
-            /* start to look if a category is set in the URL */
-            var href = window.location.href;
-            var params = href.split('/');
-            var index = 0;
-            for (var i = 0; i < this.categories.length; i++) {
-              if (this.categories[i].slug == params[params.length -1]) {
-                index = i;
-              }
-            }
-            this.$.filterControls.select("categories");
-            this.$.categories.select(index);
-          } else {
+          if (!this.categories.length) {
             this.$.filterControls.select("all");
             this.$.categories.select(-1);
           }
+          else {
+
+            if (!this.slug) {
+              for (var i = 0; i < this.categories.length; i++) {
+                if (this.categories[i].slug == this.filters[0]) {
+                  this.slug = this.categories[i].slug;
+                  this.filters.shift();
+                  break;
+                }
+              }
+            }
+
+            var CategorySelected = false;
+            if (this.slug) {
```

Comment:
Variable names should not start with a capital letter. ClassNames and EnumNames can start with a capital. variables should be camelCase. I try to follow the Mozilla and Google style guides for Javascript (Google's can be found [here](https://google.github.io/styleguide/javascriptguide.xml))